mappings/net/minecraft/client/render/entity/EntityRenderer.mapping

```diff
@@ -1,6 +1,6 @@
 CLASS net/minecraft/class_897 net/minecraft/client/render/entity/EntityRenderer
-	FIELD field_4672 shadowDarkness F
-	FIELD field_4673 shadowSize F
+	FIELD field_4672 shadowOpacity F
+	FIELD field_4673 shadowRadius F
```

Comment:
You're sure this is actually a radius (in blocks)?

Reply:
It's a radius. Not sure if in blocks. I'd have to check the code as it's been a while.

Reply:
> Not sure if in blocks.
So then it could just be a multiplier for the texture size, with no particular unit...

> It's a radius.

How can you be sure that it's a radius if you don't know the unit. For example, if this is a radius but the unit is 0.5 blocks, then it's actually a diameter with a unit of 1 block...

Reply:
Size isn't descriptive enough though. It's a radius. Not a diameter, not a circumference. It's the distance from the centre of the circle to the edge of the circle. A radius.

Reply:
I cleaned up the code to make it a little clearer to read.

```

    private static void renderShadow(MatrixStack matrices, VertexConsumerProvider vertexConsumers,
            Entity entity, float darkness, float tickDelta, WorldView world, float radius) {

        if (entity instanceof MobEntity && ((MobEntity)entity).isBaby()) {
            radius *= 0.5F;
        }

        double x = MathHelper.lerp(tickDelta, entity.lastRenderX, entity.getX());
        double y = MathHelper.lerp(tickDelta, entity.lastRenderY, entity.getY());
        double z = MathHelper.lerp(tickDelta, entity.lastRenderZ, entity.getZ());

        MatrixStack.Entry entry = matrices.peek();
        VertexConsumer vertexConsumer = vertexConsumers.getBuffer(SHADOW_LAYER);

        for (BlockPos blockPos : BlockPos.iterate(
                new BlockPos(
                        MathHelper.floor(x - radius),
                        MathHelper.floor(y - radius),
                        MathHelper.floor(z - radius)),
                new BlockPos(
                        MathHelper.floor(x + radius),
                        MathHelper.floor(y),
                        MathHelper.floor(z + radius))
        )) renderShadowPart(entry, vertexConsumer, world, blockPos, x, y, z, radius, darkness);
    }
```

```

    private static void renderShadowPart(MatrixStack.Entry matrix, VertexConsumer vertexConsumer, WorldView world,
            BlockPos pos,
            double x, double y, double z, float radius, float darkness) {
        BlockPos blockPos = pos.down();
        BlockState state = world.getBlockState(blockPos);

        if (state.getRenderType() == BlockRenderType.INVISIBLE || world.getLightLevel(pos) <= 3 || !state.isFullCube(world, blockPos)) return;

        VoxelShape shape = state.getOutlineShape(world, pos.down());

        if (shape.isEmpty()) return;

        float shadowAlpha = (float)((darkness - (y - pos.getY()) / 2.0D) * 0.5D * world.getBrightness(pos));

        if (shadowAlpha < 0) return;
        if (shadowAlpha > 1) shadowAlpha = 1;

        Box box = shape.getBoundingBox();

        float xMin = (float)(pos.getX() + box.x1 - x);
        float xMax = (float)(pos.getX() + box.x2 - x);
        float yPos = (float)(pos.getY() + box.y1 - y + 0.015625D);
        float zMin = (float)(pos.getZ() + box.z1 - z);
        float zMax = (float)(pos.getZ() + box.z2 - z);

        float u1 = -xMin / 2 / radius + 0.5F;
        float u2 = -xMax / 2 / radius + 0.5F;
        float v1 = -zMin / 2 / radius + 0.5F;
        float v2 = -zMax / 2 / radius + 0.5F;

        shadowVertex(matrix, vertexConsumer, shadowAlpha, xMin, yPos, zMin, u1, v1);
        shadowVertex(matrix, vertexConsumer, shadowAlpha, xMin, yPos, zMax, u1, v2);
        shadowVertex(matrix, vertexConsumer, shadowAlpha, xMax, yPos, zMax, u2, v2);
        shadowVertex(matrix, vertexConsumer, shadowAlpha, xMax, yPos, zMin, u2, v1);
    }
```

It's very clearly a radius of some units.